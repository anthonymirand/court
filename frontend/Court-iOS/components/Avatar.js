import React from 'react';
import PropTypes from 'prop-types';

import Mask from 'react-native-mask';
import { Image, StyleSheet, View, Platform } from 'react-native';
import IconImage from './IconImage';

/**
* Displays a Circular image of specified height
*/
export default class Avatar extends React.Component {
  render() {
    const { width, imgURL, color, animalName, showSubIcon } = this.props;
    const size = width;

    let mainAvatar = null;
    if (imgURL) {
      mainAvatar = ( <Image
                        style={{ width: size, height: size }}
                        source={{ uri: imgURL }}
                     />);
    } else if (animalName && color) {
      mainAvatar = ( <IconImage
                        avatar={ animalName }
                        size={ size }
                        color={ color }
                     />);
    } else {
      mainAvatar = ( <Image
                        style={{ width: size, height: size }}
                        source={{ uri: "http://s3.amazonaws.com/37assets/svn/765-default-avatar.png" }}
                     />);
    }

    let subAvatar = null;
    if (animalName && color && imgURL && showSubIcon) {
      subAvatar = ( <IconImage
                        avatar={ animalName }
                        size={ .3 * size }
                        color={ color }
                        showBorder={true}
                     />);
    }

    return (
      <View style={{height: size, width: size}}>
        <Mask shape={'circle'} wash>
          { mainAvatar }
        </Mask>

        {subAvatar && (
          <View style={[ styles.subAvatarWrapper, {height: .3 * size, width: .3 * size}]}>
              { subAvatar }
          </View>
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  subAvatarWrapper:{
    position: 'absolute',
    bottom: 0,
    right: 0,
    marginRight: -4,
    marginBottom: -4,
  },
});

Avatar.propTypes = {
  /**
  * Specifies diameter of resulting image
  */
  width: PropTypes.number.isRequired,
  /**
  * Flag to specify loading local image
  */
  isLocal: PropTypes.bool,
  /**
  * The string uri to fetch, either a web url, or local uri
  */
  src: PropTypes.string,
  /**
  * Background color of resulting image, if PNG
  */
  color: PropTypes.string,
};
