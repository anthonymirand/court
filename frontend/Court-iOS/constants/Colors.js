const tintColor = '#21ACA5';

export default {
  teal: '#21ACA5',
  tintColor,
  tabIconDefault: '#ccc',
  tabIconSelected: tintColor,
  tabBar: '#fefefe',
  errorBackground: 'red',
  errorText: '#fff',
  warningBackground: '#EAEB5E',
  warningText: '#666804',
  noticeBackground: tintColor,
  noticeText: '#fff',
  // To Do: remove above colors and dependencies
  lightPurple: '#9E90A2',
  coral: '#E3655B',
  mustard: '#E3B23C',
  wine: '#A4243B',
  pink: '#EA638C',
  nightSky: '#495867',
  lightBlue: '#8EB8E5',
  orange: '#EA8C55',
  peach: '#FFB997',
  darkBlue: '#28536B',
  purple: '#8783D1',
  mauve: '#BD9391',
  merlot: '#8B1E3F',
  blue: '#357DED',
  green: '#9DBF9E',
};
