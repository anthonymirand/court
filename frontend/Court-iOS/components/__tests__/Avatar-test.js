import 'react-native';
import React from 'react';
import Avatar from '../Avatar';
import Colors from '../../constants/Colors';
import renderer from 'react-test-renderer';

it('Renders Default Avatar', () => {
  const tree = renderer.create(<Avatar width={200}/>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Renders Default PNG w/o src', () => {
  const tree = renderer.create(<Avatar width={200} imgURL={null}/>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Renders Profile Picture', () => {
  const tree = renderer.create(<Avatar width={200} imgURL={"https://heightline.com/wp-content/uploads/Justin-Roberts-640x427.jpg"}/>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Renders Icon Image', () => {
  const tree = renderer.create(<Avatar width={200} imgURL={null} animalName={"sloth"} color={Colors.merlot}/>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Renders Nested Avatar', () => {
  const tree = renderer.create(<Avatar width={200} imgURL={"https://heightline.com/wp-content/uploads/Justin-Roberts-640x427.jpg"} animalName={"sloth"} color={Colors.merlot}/>).toJSON();

  expect(tree).toMatchSnapshot();
});