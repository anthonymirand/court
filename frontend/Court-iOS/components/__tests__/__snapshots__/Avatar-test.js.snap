// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Default Avatar 1`] = `
<View
  style={
    Object {
      "height": 200,
      "width": 200,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 0,
          "overflow": "hidden",
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        Object {
          "borderRadius": 0,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "http://s3.amazonaws.com/37assets/svn/765-default-avatar.png",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.03)",
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`Renders Default PNG w/o src 1`] = `
<View
  style={
    Object {
      "height": 200,
      "width": 200,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 0,
          "overflow": "hidden",
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        Object {
          "borderRadius": 0,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "http://s3.amazonaws.com/37assets/svn/765-default-avatar.png",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.03)",
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`Renders Icon Image 1`] = `
<View
  style={
    Object {
      "height": 200,
      "width": 200,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 0,
          "overflow": "hidden",
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        Object {
          "borderRadius": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#8B1E3F",
            "height": 200,
            "justifyContent": "center",
            "width": 200,
          },
          Object {},
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 120,
            "tintColor": "white",
            "width": 120,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.03)",
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`Renders Nested Avatar 1`] = `
<View
  style={
    Object {
      "height": 200,
      "width": 200,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 0,
          "overflow": "hidden",
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        Object {
          "borderRadius": 0,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://heightline.com/wp-content/uploads/Justin-Roberts-640x427.jpg",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.03)",
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`Renders Profile Picture 1`] = `
<View
  style={
    Object {
      "height": 200,
      "width": 200,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 0,
          "overflow": "hidden",
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        Object {
          "borderRadius": 0,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://heightline.com/wp-content/uploads/Justin-Roberts-640x427.jpg",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.03)",
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
